api-conversion/src/test/java/io/strimzi/kafka/api/conversion/cli/CrdUpgradeCommandIT.java

```diff
@@ -0,0 +1,213 @@
+                        .endKafka()
+                        .withNewEntityOperator()
+                            .withNewUserOperator()
+                            .endUserOperator()
+                            .withNewTopicOperator()
+                            .endTopicOperator()
+                        .endEntityOperator()
+                    .endSpec()
+                    .build();
+
+            op.inNamespace(NAMESPACE).create(kafka1);
+
+            CommandLine cmd = new CommandLine(new EntryCommand());
+            StringWriter sw = new StringWriter();
+            PrintWriter pw = new PrintWriter(sw);
+            cmd.setOut(pw);
+            cmd.setErr(pw);
+
+            int exitCode = cmd.execute("crd-upgrade");
+            assertThat(exitCode, is(0));
+
+            String commandOutput = sw.toString();
+            assertThat(commandOutput, containsString("Updating all Strimzi CRs to be stored under v1beta2"));
+            assertThat(commandOutput, containsString("Updating Kafka kafka1 to be stored as v1beta2"));
+            assertThat(commandOutput, containsString("Changing stored version in all Strimzi CRDs to v1beta2:"));
+            assertThat(commandOutput, containsString("Changing stored version in statuses of all Strimzi CRDs to v1beta2:"));
+            assertThat(commandOutput, containsString("Updating Kafka CRD"));
+            assertThat(commandOutput, containsString("Updating KafkaMirrorMaker CRD"));
+            assertThat(commandOutput, containsString("Updating KafkaRebalance CRD"));
+            assertThat(commandOutput, containsString("Updating KafkaUser CRD"));
+            assertThat(commandOutput, containsString("Updating KafkaMirrorMaker2 CRD"));
+            assertThat(commandOutput, containsString("Updating KafkaConnect CRD"));
+            assertThat(commandOutput, containsString("Updating KafkaConnectS2I CRD"));
+            assertThat(commandOutput, containsString("Updating KafkaConnector CRD"));
+            assertThat(commandOutput, containsString("Updating KafkaTopic CRD"));
+            assertThat(commandOutput, containsString("Updating KafkaBridge CRD"));
+
+            CliTestUtils.crdSpecHasUpdatedStorage(client);
+            CliTestUtils.crdStatusHasUpdatedStorageVersions(client);
+
+            Kafka actualKafka1 = op.inNamespace(NAMESPACE).withName("kafka1").get();
+            assertThat(actualKafka1, is(notNullValue()));
+            assertThat(actualKafka1.getSpec().getKafka().getListeners().getKafkaListeners(), is(nullValue()));
+            assertThat(actualKafka1.getSpec().getKafka().getListeners().getGenericKafkaListeners(), is(notNullValue()));
+            assertThat(actualKafka1.getSpec().getKafka().getListeners().getGenericKafkaListeners().size(), is(2));
+        } finally {
+            op.inNamespace(NAMESPACE).withName("kafka1").delete();
+        }
+    }
+
+    @Test
+    public void testUpgradeWithUnconvertedResourcesFails() {
```

Comment:
It's not immediately clear to me why this one should fail. A comment would be useful.